crates/lib/examples/quil_t.rs

```diff
@@ -20,17 +19,19 @@ async fn main() {
 
     let result = exe
         .compile_with_quilc(false)
-        .execute_on_qpu("Aspen-11")
+        .execute_on_qpu("Aspen-M-3")
         .await
         .expect("Program should execute successfully")
         .readout_data
-        .get_readout_values("ro".to_string(), 0)
+        .to_readout_map()
+        .expect("should be able to convert execution data to ReadoutMap")
+        .get_register_matrix("ro")
         .expect("Readout data should include 'ro'")
-        .values
-        .expect("Readout data should include values");
+        .as_integer()
+        .expect("ro should be a register of integer values")
+        .to_owned();
 
-    match result {
-        Values::IntegerValues(IntegerReadoutValues { values }) => assert!(!values.is_empty()),
-        _ => panic!("expected IntegerReadoutValues, got {:?}", result),
-    }
+    let serialized = serde_json::to_string(&result).unwrap();
+    println!("{serialized}");
+    dbg!(&result);
```

Comment:
Not the most important, but I think this fits the intent more closely. I don't think line number and file name are important for this output.

```suggestion
    println!("{result:?}");
```

Reply:
Oops, yeah, will clean up. I was doing some testing with this example and didn't fully sweep behind me.